File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
      
      map<int,int>m;
      for(int i=0;i<nums.size();i++){
        m[nums[i]]=i;
      }

      sort(nums.rbegin(),nums.rend());
      if(nums[0]>=nums[1]*2){
        return m[nums[0]];
      }else{
        return -1;
      }
    }
};

File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.dart
class Solution {
  int dominantIndex(List<int> nums) {
    Map<int, int> m = {};
  
    for (int i = 0; i < nums.length; i++) {
      m[nums[i]] = i;
    }

    nums.sort((a, b) => b.compareTo(a));

    if (nums[0] >= nums[1] * 2) {
      return m[nums[0]]!;
    } else {
      return -1;
  }
  }
}